### ISSUE TITLE
CloudBlob.AcquireLeaseAsync make proposedLeaseId optional

### ISSUE THREAD

**Issue by SeanFeldman** [Azure/azure-storage-net#241]
According to the [MSDN documentation](https://msdn.microsoft.com/en-us/library/azure/mt423343.aspx),   the overload of the [`CloudBlob.AcquireLeaseAsync` method](https://github.com/dbelcham/azure-storage-net/blob/dev_breaking/Lib/ClassLibraryCommon/Blob/CloudBlob.cs#L1867) has an optional `proposedLeaseId` which could be null if "no lease ID is proposed".

`proposedLeaseId` could be marked as optional:

```c#
public Task<string> AcquireLeaseAsync(TimeSpan? leaseTime, string proposedLeaseId = null)
```

**Comment by jehine-MSFT** [comment]
Hello @SeanFeldman ,

Thank you bringing this to our attention. We agree that this param should be made optional and have decided to make this change for our next breaking release.

Thanks for the feedback!

**Comment by jehine-MSFT** [comment]
Hello @SeanFeldman,

We have addressed this issue in the 7.0 release.

Thanks again!
